xExtension-ReadingTime/static/readingtime.js

```diff
@@ -16,49 +16,47 @@
-             }
-         }
-
+        if ("readingTime" in flux_list[i].dataset) {
+            continue;
+        }
+
+        reading_time.flux = flux_list[i];
+
+        reading_time.words_count = reading_time.flux_words_count(flux_list[i]); // count the words
+        reading_time.reading_time = reading_time.calc_read_time(reading_time.words_count, 300); // change this number (in words) to your prefered reading speed
+
+        flux_list[i].dataset.readingTime = reading_time.reading_time;
+
+        const li = document.createElement("li");
+        li.setAttribute("class", "item date");
+        li.style.width = "min-content";
+        li.style.minWidth = "40px";
+        li.style.overflow = "hidden";
+        li.style.textAlign = "right";
+        li.style.display = "table-cell";
+        li.textContent = "" + reading_time.reading_time + '\u2009m';
+ 
+        const ul = document.querySelector("#" + reading_time.flux.id + " ul.horizontal-list");
+        ul.insertBefore(li, ul.children[ul.children.length - 1]);
      }
  },
 
  flux_words_count: function flux_words_count(flux) {
 
-     reading_time.textContent = flux.querySelector('.flux_content .content').textContent; // get textContent, from the article itself (not the header, not the bottom line).
+    reading_time.textContent = flux.querySelector('.flux_content .content').textContent; // get textContent, from the article itself (not the header, not the bottom line).
 
-     // split the text to count the words correctly (source: http://www.mediacollege.com/internet/javascript/text/count-words.html)
-     reading_time.textContent = reading_time.textContent.replace(/(^\s*)|(\s*$)/gi,"");//exclude  start and end white-space
-     reading_time.textContent = reading_time.textContent.replace(/[ ]{2,}/gi," ");//2 or more space to 1
-     reading_time.textContent = reading_time.textContent.replace(/\n /,"\n"); // exclude newline with a start spacing
+    // split the text to count the words correctly (source: http://www.mediacollege.com/internet/javascript/text/count-words.html)
+    reading_time.textContent = reading_time.textContent.replace(/(^\s*)|(\s*$)/gi,"");//exclude  start and end white-space
+    reading_time.textContent = reading_time.textContent.replace(/[ ]{2,}/gi," ");//2 or more space to 1
+    reading_time.textContent = reading_time.textContent.replace(/\n /,"\n"); // exclude newline with a start spacing
 
-     return reading_time.textContent.split(' ').length;
+    return reading_time.textContent.split(' ').length;
  },
 
  calc_read_time : function calc_read_time(wd_count, speed) {
-     reading_time.read_time = Math.round(wd_count/speed);
-     if (reading_time.read_time === 0) { reading_time.read_time = '<1'; }
-     return reading_time.read_time;
+    reading_time.read_time = Math.round(wd_count/speed);
```

Comment:
The indentation seems to have changed to 3 spaces while it was 4. (Also applies to the brand new code above.)

Reply:
Consider putting an `.editorconfig` file at the root of the project to avoid such problems https://editorconfig.org as it ensures the same convention (whatever it is) throughout the project.
See e.g. https://github.com/FreshRSS/FreshRSS/blob/edge/.editorconfig

(For the record, I personally prefer tabs, to leave users the choice of how it should look like, instead of having to fight with the - less efficient in any case - 1,2,3,4,8 space conventions)

Reply:
Ah, actually if it project is being worked on from this repo, then we already do have a https://github.com/FreshRSS/Extensions/blob/master/.editorconfig
So please make sure your text editor is using it: https://editorconfig.org/#pre-installed

Reply:
Alright, good hint!

So, the file originally was indented with spaces. And the only setting for `*.js` files in the `.editorconfig` is to use tabs. Should I do another commit, re-indenting everything, or would this - for the sake of better traceability - better be handled in a separate pull request?

Reply:
Yes, better to leave unaffected lines alone unless touched or to do it separately, otherwise git blame is a pain.